include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -252,38 +254,41 @@ struct transform_reduce
         const _Size __adjusted_n = __global_offset + __n;
 
         // Coalesced load and reduce from global memory
+        union __storage
+        {
+            _Tp __v;
+            __storage() {}
+        } __res;
         if (__adjusted_global_id + __stride * __iters_per_work_item < __adjusted_n)
         {
-            typename _AccLocal::value_type __res = __unary_op(__adjusted_global_id, __acc...);
+            __res.__v = __unary_op(__adjusted_global_id, __acc...);
```

Comment:
Please, see my comment above (about explicit calls of a constructor and destructor)